dynamic/message_factory.go

```diff
@@ -159,22 +154,38 @@ func (r *KnownTypeRegistry) CreateIfKnown(messageName string) proto.Message {
 	}
 }
 
+func wellKnownType(typ reflect.Type) string {
```

Comment:
The return value is never really used. Let's instead make this `isWellKnownType(typ reflect.Type) bool`, and skip the actual invocation of the `XXX_WellKnownType` method (which wasn't there before either).

Also, instead of always allocating something here, how about something like so:

```go
func isWellKnownType(t reflect.Type) bool {
  if t.Implements(typeOfWkt) {
    return true
  }
  if msg, ok := reflect.Zero(t).Interface().(proto.Message); ok {
    name := proto.MessageName(msg)
    return wellKnownTypes[name]
  }
  return false
}
```

Reply:
Done.